students/stuartkershaw/lesson07/html_render.py

```diff
@@ -0,0 +1,207 @@
+
+            if self.kwargs:
+                self.apply_attributes(self.kwargs)
+
+            self.file_out.write(f">")
+
+            if not self.tag_name == 'a':
+                self.file_out.write(f"\n")
+
+        for el in self.content:
+            el.render(self.file_out)
+
+        if self.tag_name not in self.ignored_tag_spacing:
+            self.file_out.write(
+                self.generate_spacing(self.cur_tree_level[0])
+            )
+
+        if not self.tag_name == '':
+            self.file_out.write(f"</{self.tag_name}>")
+
+        if self.tag_name not in self.ignored_tag_spacing:
+            self.cur_tree_level[0] -= 1
+            self.file_out.write("\n")
+
+
+class Html(Element):
+    tag_name = 'html'
+
+    def render(self, file_out, cur_ind=""):
+        self.file_out = file_out
+        self.cur_ind = cur_ind
+
+        self.file_out.write('<!DOCTYPE html>\n')
+
+        Element.render(self, self.file_out, self.cur_ind)
+
+
+class Body(Element):
+    tag_name = 'body'
+
+
+class Head(Element):
+    tag_name = 'head'
+
+
+class OneLineTag(Element):
+    def render(self, file_out, cur_ind="", level=None):
+        self.file_out = file_out
+        self.cur_ind = cur_ind
+        self.level = level
+
+        self.file_out.write(
+            self.generate_spacing(self.cur_tree_level[0] + 1)
+        )
+        self.file_out.write(f"<{self.tag_name}")
+
+        if self.kwargs:
+            if not self.tag_name == "h":
+                self.apply_attributes(self.kwargs)
+            else:
+                self.file_out.write(f"{self.kwargs['level']}")
+
+        self.file_out.write(f">")
+
+        for el in self.content:
+            el.render(self.file_out)
+
+        self.file_out.write(f"</{self.tag_name}")
+
+        if self.kwargs.get("level"):
+            self.file_out.write(f"{self.kwargs.get('level')}")
+
+        self.file_out.write(">\n")
+
+
+class Title(OneLineTag):
+    tag_name = 'title'
+
+
+class P(OneLineTag):
+    tag_name = 'p'
+
+
+class SelfClosingTag(Element):
+    def render(self, file_out, cur_ind=""):
+        self.file_out = file_out
+        self.cur_ind = cur_ind
+
+        self.file_out.write(
+            self.generate_spacing(self.cur_tree_level[0] + 1)
+        )
+
+        self.file_out.write(f"<{self.tag_name}")
+
+        if self.kwargs:
+            self.apply_attributes(self.kwargs)
+
+        try:
+            if len(self.content):
+                raise TypeError
+            self.file_out.write(f" />\n")
+        except TypeError:
+            print('This element does not accept nested content.')
+
+
+class Hr(SelfClosingTag):
+    tag_name = 'hr'
+
+
+class Br(SelfClosingTag):
+    tag_name = 'br'
+
+
+class A(Element):
+    tag_name = 'a'
+
+    def __init__(self, link, content):
+        self.link = link
+        self.content = content
+        self.attributes = {'href': self.link}
+
+        Element.__init__(self, content=self.content, **self.attributes)
```

Comment:
`super().__init__` call


you don't need any of the instance var definitions there:
```
def __init__(self, link, content):
    super().__init__(content=content, **{'href': link})
````